client/src/pages/AccountSettingsPage.jsx

```diff
@@ -0,0 +1,320 @@
+      setPassword(userInfo.password);
+      setConfirmPassword(userInfo.password);
+    } catch {
+      setErrorMessage(
+        'There was a problem loading your user information. Please try agian'
+      );
+    }
+  };
+
+  useEffect(() => {
+    fetchUserInfo();
+  }, []);
+
+  return (
+    <BasePage>
+      <BackgroundShapes />
+      <Grid zIndex={0} container justifyContent="center" alignItems="center">
+        <Grid item xs={12} lg={10}>
+          <Card>
+            <Typography variant="pageTitle">Account Settings</Typography>
+            {rowFields.map((row) => {
+              const canEdit = editableFields[row.editableFieldKey];
+              const preview = row.previewModeFields;
+              const edit = row.editModeFields;
+              const editFieldsFlex = 12 / (edit.fields.length + 1);
+
+              return (
+                <Grid
+                  container
+                  key={row.editableFieldKey}
+                  pt="1.5rem"
+                  alignItems="center"
+                  spacing={4}
+                >
+                  {canEdit ? (
+                    <>
+                      {edit.fields.map((field) => (
+                        <Grid
+                          item
+                          sm={editFieldsFlex}
+                          xs={12}
+                          key={field.label}
+                        >
+                          {field.isHidden ? (
+                            <TextField // replace with HiddenTextComponent
+                              value={field.value}
+                              label={field.label}
+                              fullWidth
+                              onChange={field.onChange}
+                            />
+                          ) : (
+                            <TextField
+                              value={field.value}
+                              label={field.label}
+                              fullWidth
+                              onChange={field.onChange}
+                            />
+                          )}
+                        </Grid>
+                      ))}
+                      <Grid
+                        container
+                        item
+                        sm={editFieldsFlex}
+                        xs={12}
+                        justifyContent={{ sm: 'flex-end', xs: 'flex-start' }}
+                      >
+                        <Button
+                          variant="transparent"
+                          sx={{
+                            borderRadius: '2rem',
+                            borderWidth: '1px',
+                            borderStyle: 'solid',
+                            borderColor: 'secondary.main',
+                            color: 'secondary.main',
+                          }}
+                          onClick={async () => {
+                            const errorMessage = await edit.getErrorMessage();
```

Comment:
(follow up) here you could wrap the function in a try/catch to get the error message instead of checking if error message is null 

functionally it's the same so if you think this is fine then just leave it lol